lib/http2/connection.c

```diff
@@ -1353,6 +1359,9 @@ static void push_path(h2o_req_t *src_req, const char *abspath, size_t abspath_le
     h2o_http2_conn_t *conn = (void *)src_req->conn;
     h2o_http2_stream_t *src_stream = H2O_STRUCT_FROM_MEMBER(h2o_http2_stream_t, req, src_req);
 
+    if (conn->state == H2O_HTTP2_CONN_STATE_IS_CLOSING)
+        return;
+
```

Comment:
Generally speaking, do we do these checks in the code that sends data?

Assuming that we do not, I think we should omit making such checks unless it's required for other reasons.

Reply:
no need!